iree/compiler/Conversion/Common/LinalgBufferizePass.cpp

```diff
@@ -495,13 +495,19 @@ LogicalResult convertInterfaceStoreTensorOp(
     BlockAndValueMapping &bvm) {
   OpBuilder::InsertionGuard g(b);
   b.setInsertionPoint(storeOp);
-  Value storeTo = bvm.lookup(storeOp.target());
-  // If the value already has a mapping, it should already have been updated in
-  // place by the converted producer.
-  if (storeTo) {
+
+  // If we have both the source and target buffer pointing to the same binding,
+  // then it's an indication that we are performing in-place update. For such
+  // cases, we can just remove this store.
+  auto storeTo = bvm.lookup(storeOp.target())
+                     .getDefiningOp<IREE::HAL::InterfaceBindingSubspanOp>();
+  auto storeFrom = bvm.lookup(storeOp.value())
```

Comment:
The value could also be a subview (like I mentioned in the previous comment). See `preProcessInterfaceStoreTensorOp`

Reply:
Actually on second thought, this might be safer. Lets go with this for now and revisit later.